homeassistant/components/light/ads.py

```diff
@@ -0,0 +1,119 @@
+"""Support for ADS light sources."""
+
+import logging
+import math
+
+import voluptuous as vol
+
+from homeassistant.components.light import Light, ATTR_BRIGHTNESS, \
+    SUPPORT_BRIGHTNESS, PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME
+from homeassistant.components.ads import DATA_ADS, CONF_ADSVAR
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+DEPENDENCIES = ['ads']
+DEFAULT_NAME = 'ADS Light'
+CONF_ADSVAR_BRIGHTNESS = 'adsvar_brightness'
+SUPPORT_ADS = SUPPORT_BRIGHTNESS
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_ADSVAR): cv.string,
+    vol.Optional(CONF_ADSVAR_BRIGHTNESS): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the light platform for ADS."""
+    ads_hub = hass.data.get(DATA_ADS)
+    if not ads_hub:
+        return False
+
+    varname_enable = config.get(CONF_ADSVAR)
+    varname_brightness = config.get(CONF_ADSVAR_BRIGHTNESS)
+    devname = config.get(CONF_NAME)
+
+    add_devices([AdsLight(ads_hub, varname_enable, varname_brightness,
+                          devname)], True)
+
+
+class AdsLight(Light):
+    """Representation of ADS light."""
+
+    def __init__(self, ads_hub, varname_enable, varname_brightness, devname):
+        """Initialize AdsLight entity."""
+        self._ads_hub = ads_hub
+        self._on_state = False
+        self._brightness = 50
+        self._devname = devname
+        self.varname_enable = varname_enable
+        self.varname_brightness = varname_brightness
+        self.stype = 'dimmer'
+
+    @property
+    def name(self):
+        """Return the name of the device if any."""
+        return self._devname
+
+    @property
+    def brightness(self):
+        """
+        Brightness of the light.
+
+        This method is optional. Removing it indicates to Home Assistant
```

Comment:
This used to be true, but now I think it's supported features that should indicate this. I'm not :100: though, if the frontend is totally updated yet.

Reply:
This comment was just copy&past. I removed it.